src/simulation/task_process_simulation_outputs.py

```diff
@@ -20,51 +18,49 @@
 ]
 
 
-def get_available_scenarios(named_scenarios):
-    available_scenarios = {
-        name for name, spec in named_scenarios.items() if spec["n_seeds"] > 0
-    }
-    return available_scenarios
-
-
-def create_path_for_weekly_outcome_of_scenario(name, fast_flag, outcome, groupby):
-    if groupby is None:
-        file_name = f"{fast_flag}_{name}_{outcome}.pkl"
-    else:
-        file_name = f"{fast_flag}_{name}_{outcome}_by_{groupby}.pkl"
-    return BLD / "simulations" / "incidences" / file_name
-
-
-def create_incidence_parametrization(named_scenarios, fast_flag, outcomes):
+def _create_create_weekly_incidence_parametrization(outcomes):
+    named_scenarios = get_named_scenarios()
     parametrization = []
     for outcome in outcomes:
         for groupby in [None, "age_group_rki"]:
             for name, specs in named_scenarios.items():
                 dependencies = {
-                    seed: create_path_to_last_states_of_simulation(
-                        fast_flag, name, seed
-                    )
+                    seed: create_path_to_last_states_of_simulation(name, seed)
                     for seed in range(specs["n_seeds"])
                 }
 
                 # this handles the case of 0 seeds, i.e. skipped scenarios
                 if dependencies:
-                    produces = create_path_for_weekly_outcome_of_scenario(
-                        name, fast_flag, outcome, groupby
+                    produces = create_path_to_weekly_outcome_of_scenario(
+                        name, outcome, groupby
                     )
 
                     parametrization.append((dependencies, outcome, groupby, produces))
 
     return "depends_on, outcome, groupby, produces", parametrization
 
 
-_SIGNATURE, _PARAMETRIZATION = create_incidence_parametrization(
-    NAMED_SCENARIOS, FAST_FLAG, OUTCOMES
-)
+def _create_calculate_share_known_cases_of_scenarios_parametrization():
+    """Create the parametrization for the share known cases."""
+    named_scenarios = get_named_scenarios()
+    available_scenarios = get_available_scenarios(named_scenarios)
+    parametrization = []
+    outcomes = ["currently_infected", "knows_currently_infected"]
+
+    for scenario_name in available_scenarios:
+        depends_on = {
+            outcome: create_path_to_weekly_outcome_of_scenario(
+                scenario_name, outcome, "age_group_rki"
+            )
+            for outcome in outcomes
+        }
+        produces = create_path_to_share_known_cases_of_scenario(scenario_name)
+        parametrization.append((depends_on, produces))
+    return "depends_on, produces", parametrization
 
 
 @pytask.mark.depends_on({"calculate_weekly_incidences": SRC / "calculate_moments.py"})
-@pytask.mark.parametrize(_SIGNATURE, _PARAMETRIZATION)
+@pytask.mark.parametrize(*_create_create_weekly_incidence_parametrization(OUTCOMES))
```

Comment:
I would say its better to have `_signature` and `_parametrization` to be able to debug, because it is hard to debug returns immediately. But does not need a change. Just for discussion.

Reply:
Yes, that's correct. I'll keep it in mind in the future. Thanks!